### ISSUE TITLE
Installation failing when following instruction

### ISSUE THREAD

**Issue by gggh000** [microsoft/superbenchmark#293]
```
Ubuntu1804


nonroot@nonroot-MS-7B22:~/superbenchmark$ pwd
/home/nonroot/superbenchmark
nonroot@nonroot-MS-7B22:~/superbenchmark$ git remote -v
origin	https://github.com/microsoft/superbenchmark (fetch)
origin	https://github.com/microsoft/superbenchmark (push)
nonroot@nonroot-MS-7B22:~/superbenchmark$ 

nonroot@nonroot-MS-7B22:~/superbenchmark$ sudo python3 -m pip install .
Keyring is skipped due to an exception: org.freedesktop.DBus.Error.NoServer: Failed to connect to socket /tmp/dbus-9Lms1YHOlb: Connection refused
WARNING: The directory '/home/nonroot/.cache/pip' or its parent directory is not owned or is not writable by the current user. The cache has been disabled. Check the permissions and owner of that directory. If executing pip with sudo, you should use sudo's -H flag.
Processing /home/nonroot/superbenchmark
  Preparing metadata (setup.py) ... done
Collecting ansible_runner>=2.0.0rc1
  Downloading ansible_runner-2.1.1-py3-none-any.whl (83 kB)
     |████████████████████████████████| 83 kB 4.7 MB/s             
Collecting colorlog>=4.7.2
  Downloading colorlog-6.6.0-py2.py3-none-any.whl (11 kB)
Collecting jinja2>=2.10.1
  Downloading Jinja2-3.0.3-py3-none-any.whl (133 kB)
     |████████████████████████████████| 133 kB 12.2 MB/s            
Requirement already satisfied: joblib>=1.0.1 in /usr/local/lib/python3.6/dist-packages (from superbench==0.4.0) (1.1.0)
Collecting jsonlines>=2.0.0
  Downloading jsonlines-3.0.0-py3-none-any.whl (8.5 kB)
Collecting knack>=0.7.2
  Downloading knack-0.9.0-py3-none-any.whl (59 kB)
     |████████████████████████████████| 59 kB 94.4 MB/s            
Requirement already satisfied: matplotlib>=3.0.0 in /usr/local/lib/python3.6/dist-packages (from superbench==0.4.0) (3.3.4)
Collecting natsort>=7.1.1
  Downloading natsort-8.0.2-py3-none-any.whl (37 kB)
Collecting omegaconf==2.0.6
  Downloading omegaconf-2.0.6-py3-none-any.whl (36 kB)
Collecting openpyxl>=3.0.7
  Downloading openpyxl-3.0.9-py2.py3-none-any.whl (242 kB)
     |████████████████████████████████| 242 kB 44.7 MB/s            
Requirement already satisfied: pandas>=1.1.5 in /usr/local/lib/python3.6/dist-packages (from superbench==0.4.0) (1.1.5)
Collecting pyyaml>=5.3
  Downloading PyYAML-6.0-cp36-cp36m-manylinux_2_5_x86_64.manylinux1_x86_64.manylinux_2_12_x86_64.manylinux2010_x86_64.whl (603 kB)
     |████████████████████████████████| 603 kB 59.7 MB/s            
Collecting seaborn>=0.11.2
  Downloading seaborn-0.11.2-py3-none-any.whl (292 kB)
     |████████████████████████████████| 292 kB 62.9 MB/s            
Collecting tcping>=0.1.1rc1
  Downloading tcping-0.1.1rc1.tar.gz (4.1 kB)
  Preparing metadata (setup.py) ... done
Collecting xlrd>=2.0.1
  Downloading xlrd-2.0.1-py2.py3-none-any.whl (96 kB)
     |████████████████████████████████| 96 kB 45.4 MB/s            
Collecting xlsxwriter>=1.3.8
  Downloading XlsxWriter-3.0.2-py3-none-any.whl (149 kB)
     |████████████████████████████████| 149 kB 69.3 MB/s            
Collecting xmltodict>=0.12.0
  Downloading xmltodict-0.12.0-py2.py3-none-any.whl (9.2 kB)
Collecting ansible_base>=2.10.9
  Downloading ansible-base-2.10.16.tar.gz (6.1 MB)
     |████████████████████████████████| 6.1 MB 25.8 MB/s            
  Preparing metadata (setup.py) ... done
Requirement already satisfied: dataclasses in /home/nonroot/.local/lib/python3.6/site-packages (from omegaconf==2.0.6->superbench==0.4.0) (0.8)
Requirement already satisfied: typing-extensions in /home/nonroot/.local/lib/python3.6/site-packages (from omegaconf==2.0.6->superbench==0.4.0) (4.0.1)
Requirement already satisfied: cryptography in /usr/lib/python3/dist-packages (from ansible_base>=2.10.9->superbench==0.4.0) (2.1.4)
Collecting packaging
  Downloading packaging-21.3-py3-none-any.whl (40 kB)
     |████████████████████████████████| 40 kB 72.5 MB/s            
Requirement already satisfied: six in /home/nonroot/.local/lib/python3.6/site-packages (from ansible_runner>=2.0.0rc1->superbench==0.4.0) (1.16.0)
Collecting pexpect>=4.5
  Downloading pexpect-4.8.0-py2.py3-none-any.whl (59 kB)
     |████████████████████████████████| 59 kB 63.9 MB/s            
Collecting python-daemon
  Downloading python_daemon-2.3.0-py2.py3-none-any.whl (35 kB)
Requirement already satisfied: MarkupSafe>=2.0 in /usr/local/lib/python3.6/dist-packages (from jinja2>=2.10.1->superbench==0.4.0) (2.0.1)
Collecting attrs>=19.2.0
  Downloading attrs-21.4.0-py2.py3-none-any.whl (60 kB)
     |████████████████████████████████| 60 kB 63.3 MB/s            
Collecting argcomplete
  Downloading argcomplete-2.0.0-py2.py3-none-any.whl (37 kB)
Collecting jmespath
  Downloading jmespath-0.10.0-py2.py3-none-any.whl (24 kB)
Requirement already satisfied: tabulate in /usr/local/lib/python3.6/dist-packages (from knack>=0.7.2->superbench==0.4.0) (0.8.9)
Collecting pygments
  Downloading Pygments-2.11.2-py3-none-any.whl (1.1 MB)
     |████████████████████████████████| 1.1 MB 99.9 MB/s            
Requirement already satisfied: python-dateutil>=2.1 in /usr/local/lib/python3.6/dist-packages (from matplotlib>=3.0.0->superbench==0.4.0) (2.8.2)
Requirement already satisfied: cycler>=0.10 in /usr/local/lib/python3.6/dist-packages (from matplotlib>=3.0.0->superbench==0.4.0) (0.11.0)
Requirement already satisfied: kiwisolver>=1.0.1 in /usr/local/lib/python3.6/dist-packages (from matplotlib>=3.0.0->superbench==0.4.0) (1.3.1)
Requirement already satisfied: numpy>=1.15 in /home/nonroot/.local/lib/python3.6/site-packages (from matplotlib>=3.0.0->superbench==0.4.0) (1.19.5)
Requirement already satisfied: pillow>=6.2.0 in /usr/local/lib/python3.6/dist-packages (from matplotlib>=3.0.0->superbench==0.4.0) (8.4.0)
Requirement already satisfied: pyparsing!=2.0.4,!=2.1.2,!=2.1.6,>=2.0.3 in /usr/local/lib/python3.6/dist-packages (from matplotlib>=3.0.0->superbench==0.4.0) (3.0.7)
Collecting et-xmlfile
  Downloading et_xmlfile-1.1.0-py3-none-any.whl (4.7 kB)
Requirement already satisfied: pytz>=2017.2 in /usr/lib/python3/dist-packages (from pandas>=1.1.5->superbench==0.4.0) (2018.3)
Requirement already satisfied: scipy>=1.0 in /usr/local/lib/python3.6/dist-packages (from seaborn>=0.11.2->superbench==0.4.0) (1.5.4)
Collecting click
  Downloading click-8.0.3-py3-none-any.whl (97 kB)
     |████████████████████████████████| 97 kB 64.2 MB/s            
Collecting prettytable
  Downloading prettytable-2.5.0-py3-none-any.whl (24 kB)
Requirement already satisfied: ptyprocess>=0.5 in /usr/local/lib/python3.6/dist-packages (from pexpect>=4.5->ansible_runner>=2.0.0rc1->superbench==0.4.0) (0.7.0)
Requirement already satisfied: importlib-metadata<5,>=0.23 in /usr/local/lib/python3.6/dist-packages (from argcomplete->knack>=0.7.2->superbench==0.4.0) (4.8.3)
Requirement already satisfied: wcwidth in /usr/local/lib/python3.6/dist-packages (from prettytable->tcping>=0.1.1rc1->superbench==0.4.0) (0.2.5)
Requirement already satisfied: docutils in /usr/local/lib/python3.6/dist-packages (from python-daemon->ansible_runner>=2.0.0rc1->superbench==0.4.0) (0.18.1)
Requirement already satisfied: lockfile>=0.10 in /usr/local/lib/python3.6/dist-packages (from python-daemon->ansible_runner>=2.0.0rc1->superbench==0.4.0) (0.12.2)
Requirement already satisfied: setuptools in /usr/lib/python3/dist-packages (from python-daemon->ansible_runner>=2.0.0rc1->superbench==0.4.0) (39.0.1)
Requirement already satisfied: zipp>=0.5 in /usr/local/lib/python3.6/dist-packages (from importlib-metadata<5,>=0.23->argcomplete->knack>=0.7.2->superbench==0.4.0) (3.6.0)
Building wheels for collected packages: superbench, ansible-base, tcping
  Building wheel for superbench (setup.py) ... done
  Created wheel for superbench: filename=superbench-0.4.0-py3-none-any.whl size=139599 sha256=f08e0e47cd18aadcd92ad96228b346a44daa7d569eb24ad2269e8d3ebb472819
  Stored in directory: /tmp/pip-ephem-wheel-cache-dsy5_fgx/wheels/dc/bb/c9/0181c21c034d8eb5089301c92e8e8b249ecc42b0a8569ef352
  Building wheel for ansible-base (setup.py) ... done
  Created wheel for ansible-base: filename=ansible_base-2.10.16-py3-none-any.whl size=1871330 sha256=550d11df3e25bf809017d59c36152caf22db6031f3823c725064ac06325d9600
  Stored in directory: /tmp/pip-ephem-wheel-cache-dsy5_fgx/wheels/ac/3a/eb/1953c987dfe9515f0b3c0770e22520361beedf030ec746b716
  Building wheel for tcping (setup.py) ... done
  Created wheel for tcping: filename=tcping-0.1.1rc1-py3-none-any.whl size=6400 sha256=8e0a41e98d6c0f4d7ea34b176f95b1cc7444edd7eeddf591d3a7fcc948d4f381
  Stored in directory: /tmp/pip-ephem-wheel-cache-dsy5_fgx/wheels/79/b4/79/cd1464d78ff94847f17dde162e88301861ffcdbae7b57279f0
Successfully built superbench ansible-base tcping
Installing collected packages: pyyaml, python-daemon, pygments, prettytable, pexpect, packaging, jmespath, jinja2, et-xmlfile, click, attrs, argcomplete, xmltodict, xlsxwriter, xlrd, tcping, seaborn, openpyxl, omegaconf, natsort, knack, jsonlines, colorlog, ansible-runner, ansible-base, superbench
  Attempting uninstall: pyyaml
    Found existing installation: PyYAML 3.12
ERROR: Cannot uninstall 'PyYAML'. It is a distutils installed project and thus we cannot accurately determine which files belong to it which would lead to only a partial uninstall.
nonroot@nonroot-MS-7B22:~/superbenchmark$ sudo pip3 install --upgrade PyYAML
WARNING: The directory '/home/nonroot/.cache/pip' or its parent directory is not owned or is not writable by the current user. The cache has been disabled. Check the permissions and owner of that directory. If executing pip with sudo, you should use sudo's -H flag.
Requirement already satisfied: PyYAML in /usr/lib/python3/dist-packages (3.12)
Collecting PyYAML
  Downloading PyYAML-6.0-cp36-cp36m-manylinux_2_5_x86_64.manylinux1_x86_64.manylinux_2_12_x86_64.manylinux2010_x86_64.whl (603 kB)
     |████████████████████████████████| 603 kB 4.2 MB/s            
Installing collected packages: PyYAML
  Attempting uninstall: PyYAML
    Found existing installation: PyYAML 3.12
ERROR: Cannot uninstall 'PyYAML'. It is a distutils installed project and thus we cannot accurately determine which files belong to it which would lead to only a partial uninstall.
nonroot@nonroot-MS-7B22:~/superbenchmark$ sudo python3 -m pip install .^C
nonroot@nonroot-MS-7B22:~/superbenchmark$ pwd
/home/nonroot/superbenchmark
nonroot@nonroot-MS-7B22:~/superbenchmark$ git remote -v
origin	https://github.com/microsoft/superbenchmark (fetch)
origin	https://github.com/microsoft/superbenchmark (push)
nonroot@nonroot-MS-7B22:~/superbenchmark$ 

```

**Comment by abuccts** [comment]
try with `pip install --ignore-installed ...`, here's a [reference](https://stackoverflow.com/a/53534728/7630831)

or you are recommended to use [`venv`](https://packaging.python.org/en/latest/guides/installing-using-pip-and-virtual-environments/#creating-a-virtual-environment) in this case:
```sh
python3 -m venv ~/venv
source ~/venv/bin/activate

cd ~/superbenchmark
sudo python3 -m pip install .
```

**Comment by gggh000** [comment]
venv had same error.

**Comment by gggh000** [comment]
ble_base-2.10.16-py3-none-any.whl size=1871330 sha256=ca224bd24216cd748af5085cfdbe39e46b8142e34077fe6d6a8bc141a4665f57
  Stored in directory: /tmp/pip-ephem-wheel-cache-4fa7plan/wheels/ac/3a/eb/1953c987dfe9515f0b3c0770e22520361beedf030ec746b716
  Building wheel for tcping (setup.py) ... done
  Created wheel for tcping: filename=tcping-0.1.1rc1-py3-none-any.whl size=6400 sha256=5d1509db1fe759333298fb7a492af9746f57d7feba418a8b48fc038f40789502
  Stored in directory: /tmp/pip-ephem-wheel-cache-4fa7plan/wheels/79/b4/79/cd1464d78ff94847f17dde162e88301861ffcdbae7b57279f0
Successfully built superbench ansible-base tcping
Installing collected packages: pyyaml, python-daemon, pygments, prettytable, pexpect, packaging, jmespath, jinja2, et-xmlfile, click, attrs, argcomplete, xmltodict, xlsxwriter, xlrd, tcping, seaborn, openpyxl, omegaconf, natsort, knack, jsonlines, colorlog, ansible-runner, ansible-base, superbench
  Attempting uninstall: pyyaml
    Found existing installation: PyYAML 3.12
ERROR: Cannot uninstall 'PyYAML'. It is a distutils installed project and thus we cannot accurately determine which files belong to it which would lead to only a partial uninstall.
(venv) nonroot@nonroot-MS-7B22:~/superbenchmark$ 
(venv) nonroot@nonroot-MS-7B22:~/superbenchmark$

**Comment by gggh000** [comment]
ignore insatllation does not work, please reproduce on your own end before giving needless advice becaue it is wasting my time and it is not my job to do a qa
```

(venv) nonroot@nonroot-MS-7B22:~/superbenchmark$ sudo python3 -m pip install --ignore-installed .

    |████████████████████████████████| 570 kB 59.3 MB/s            
Collecting lockfile>=0.10
  Downloading lockfile-0.12.2-py2.py3-none-any.whl (13 kB)
Collecting setuptools
  Downloading setuptools-59.6.0-py3-none-any.whl (952 kB)
     |████████████████████████████████| 952 kB 73.6 MB/s            
Collecting pycparser
  Downloading pycparser-2.21-py2.py3-none-any.whl (118 kB)
     |████████████████████████████████| 118 kB 65.2 MB/s            
Collecting zipp>=0.5
  Downloading zipp-3.6.0-py3-none-any.whl (5.3 kB)
Building wheels for collected packages: superbench, ansible-base, tcping
  Building wheel for superbench (setup.py) ... done
  Created wheel for superbench: filename=superbench-0.4.0-py3-none-any.whl size=139599 sha256=841f34e1806abb0f35d92a23774f68321ed37dd1ac07ffa4a09d6e23789683eb
  Stored in directory: /tmp/pip-ephem-wheel-cache-mfa9zcos/wheels/dc/bb/c9/0181c21c034d8eb5089301c92e8e8b249ecc42b0a8569ef352
  Building wheel for ansible-base (setup.py) ... done
  Created wheel for ansible-base: filename=ansible_base-2.10.16-py3-none-any.whl size=1871330 sha256=6d1a9c549f60b27c50abe0226f0e132a7a9ce69135696bf05e8e1df3de710616
  Stored in directory: /tmp/pip-ephem-wheel-cache-mfa9zcos/wheels/ac/3a/eb/1953c987dfe9515f0b3c0770e22520361beedf030ec746b716
  Building wheel for tcping (setup.py) ... done
  Created wheel for tcping: filename=tcping-0.1.1rc1-py3-none-any.whl size=6400 sha256=1b468839b92f2ccdd44ac9ae2acea3c58520436114a5ca7c0f9a993502e7db44
  Stored in directory: /tmp/pip-ephem-wheel-cache-mfa9zcos/wheels/79/b4/79/cd1464d78ff94847f17dde162e88301861ffcdbae7b57279f0
Successfully built superbench ansible-base tcping
Installing collected packages: zipp, typing-extensions, six, pycparser, wcwidth, setuptools, pytz, python-dateutil, pyparsing, ptyprocess, pillow, numpy, MarkupSafe, lockfile, kiwisolver, importlib-metadata, docutils, cycler, cffi, tabulate, scipy, pyyaml, python-daemon, pygments, prettytable, pexpect, pandas, packaging, matplotlib, jmespath, jinja2, et-xmlfile, dataclasses, cryptography, click, attrs, argcomplete, xmltodict, xlsxwriter, xlrd, tcping, seaborn, openpyxl, omegaconf, natsort, knack, jsonlines, joblib, colorlog, ansible-runner, ansible-base, superbench
ERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.
launchpadlib 1.10.6 requires testresources, which is not installed.
Successfully installed MarkupSafe-2.0.1 ansible-base-2.10.16 ansible-runner-2.1.1 argcomplete-2.0.0 attrs-21.4.0 cffi-1.15.0 click-8.0.3 colorlog-6.6.0 cryptography-36.0.1 cycler-0.11.0 dataclasses-0.8 docutils-0.18.1 et-xmlfile-1.1.0 importlib-metadata-4.8.3 jinja2-3.0.3 jmespath-0.10.0 joblib-1.1.0 jsonlines-3.0.0 kiwisolver-1.3.1 knack-0.9.0 lockfile-0.12.2 matplotlib-3.3.4 natsort-8.1.0 numpy-1.19.5 omegaconf-2.0.6 openpyxl-3.0.9 packaging-21.3 pandas-1.1.5 pexpect-4.8.0 pillow-8.4.0 prettytable-2.5.0 ptyprocess-0.7.0 pycparser-2.21 pygments-2.11.2 pyparsing-3.0.7 python-daemon-2.3.0 python-dateutil-2.8.2 pytz-2021.3 pyyaml-6.0 scipy-1.5.4 seaborn-0.11.2 setuptools-59.6.0 six-1.16.0 superbench-0.4.0 tabulate-0.8.9 tcping-0.1.1rc1 typing-extensions-4.0.1 wcwidth-0.2.5 xlrd-2.0.1 xlsxwriter-3.0.2 xmltodict-0.12.0 zipp-3.6.0
WARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager. It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv

```

**Comment by abuccts** [comment]
First of all, I didn't ask you to use or install this project, you did it by your own because this is an open source project. But **Open Source does not mean "Includes Free Support"**, I do not have duty to solve your problem. It's your own business to ask question and solve your own problem. And please be respect to other's time cause you didn't pay it.

Second, I cannot reproduce your issue nor can I know what's wrong with your environment. What I did is helping you search the similar issues using the limited log you posted, you should do it by your own at the beginning.

In summary, your problem is cannot install `pyyaml` in pip because your environment has a system conflict. This is off-topic to this project and you'd better to ask somewhere else. Here's an instruction for you in case you don't know [how to ask questions](https://stackoverflow.com/help/how-to-ask).
